File: mobile/lib/features/task/domain/enum/task_priority.dart
enum TaskPriority { low, medium, high, critical }

extension TaskPriorityX on TaskPriority {
  static TaskPriority fromApi(String? value) {
    switch ((value ?? '').trim().toUpperCase()) {
      case 'LOW':
        return TaskPriority.low;
      case 'HIGH':
        return TaskPriority.high;
      case 'CRITICAL':
        return TaskPriority.critical;
      case 'MEDIUM':
      default:
        return TaskPriority.medium;
    }
  }

  String get apiValue {
    switch (this) {
      case TaskPriority.low:
        return 'LOW';
      case TaskPriority.medium:
        return 'MEDIUM';
      case TaskPriority.high:
        return 'HIGH';
      case TaskPriority.critical:
        return 'CRITICAL';
    }
  }

  String get label {
    switch (this) {
      case TaskPriority.low:
        return 'Low';
      case TaskPriority.medium:
        return 'Medium';
      case TaskPriority.high:
        return 'High';
      case TaskPriority.critical:
        return 'Critical';
    }
  }
}

File: backend/src/main/java/com/lifeos/backend/task/domain/enums/TaskPriority.java
package com.lifeos.backend.task.domain.enums;

/**
 * Business priority of a task.
 *
 * This affects sorting, Today focus, notification urgency,
 * and later scoring/AI suggestions.
 */
public enum TaskPriority {

    /**
     * Must be handled soon.
     * Example: exam submission, payment deadline, urgent production bug.
     */
    CRITICAL(0),

    /**
     * Important but not emergency-level.
     * Example: project milestone, important study block.
     */
    HIGH(1),

    /**
     * Normal default priority.
     */
    MEDIUM(2),

    /**
     * Low-pressure task.
     * Example: optional cleanup, nice-to-have improvement.
     */
    LOW(3);

    private final int rank;

    TaskPriority(int rank) {
        this.rank = rank;
    }

    public int rank() {
        return rank;
    }

    public boolean isHighUrgency() {
        return this == CRITICAL || this == HIGH;
    }

    public boolean isLowerThan(TaskPriority other) {
        if (other == null) {
            return false;
        }

        return this.rank > other.rank;
    }

    public boolean isHigherThan(TaskPriority other) {
        if (other == null) {
            return true;
        }

        return this.rank < other.rank;
    }
}